bridge-deploy/bridge_deploy/home.py

```diff
@@ -0,0 +1,118 @@
+import json
+from typing import Dict, NamedTuple
+from web3.contract import Contract
+from deploy_tools.deploy import (
+    deploy_compiled_contract,
+    increase_transaction_options_nonce,
+    send_function_call_transaction,
+)
+
+
+class DeployedHomeBridgeResult(NamedTuple):
+    home_bridge: Contract
+    home_bridge_block_number: int
+
+
+def load_contract(contract_name):
+    with open(
+        f"../poa-bridge-contracts/build/contracts/{contract_name}.json"
+    ) as json_file:
+        return json.load(json_file)
+
+
+def deploy_home_bridge_contract(
+    *, web3, transaction_options: Dict = None, private_key=None
+):
+
+    if transaction_options is None:
+        transaction_options = {}
+
+    latest_block = web3.eth.getBlock("latest")
+
+    # Deploy home bridge proxy
+    eternal_storage_proxy_src = load_contract("EternalStorageProxy")
+    home_bridge_storage_contract: Contract = deploy_compiled_contract(
+        abi=eternal_storage_proxy_src["abi"],
+        bytecode=eternal_storage_proxy_src["bytecode"],
+        web3=web3,
+        transaction_options=transaction_options,
+        private_key=private_key,
+    )
+    increase_transaction_options_nonce(transaction_options)
+
+    # Deploy home bridge implementation
+    home_bridge_src = load_contract("HomeBridgeErcToNative")
+    home_bridge_contract: Contract = deploy_compiled_contract(
+        abi=home_bridge_src["abi"],
+        bytecode=home_bridge_src["bytecode"],
+        web3=web3,
+        transaction_options=transaction_options,
+        private_key=private_key,
+    )
+    increase_transaction_options_nonce(transaction_options)
+
+    # Connect home bridge proxy to implementation
+    home_bridge_storage_upgrade = home_bridge_storage_contract.functions.upgradeTo(
+        1, home_bridge_contract.address
+    )
+    send_function_call_transaction(
+        home_bridge_storage_upgrade,
+        web3=web3,
+        transaction_options=transaction_options,
+        private_key=private_key,
+    )
+    increase_transaction_options_nonce(transaction_options)
+
+    # Use proxy from now on
+    home_bridge_contract = web3.eth.contract(
+        abi=home_bridge_src["abi"], address=home_bridge_storage_contract.address
+    )
+
+    # TODO: Initialize bridge
+
+    # TODO: transfer ownership
+
+    contracts = DeployedHomeBridgeResult(home_bridge_contract, latest_block.number)
```

Comment:
I still believe it is weird to return the block at which the contract was deployed, and do not see where you use it except for printing it.

Reply:
The bridge workers need it to limit the number of blocks in which they look for events.

Reply:
Exactly. We could get it from the bridge contract itself later on, I think, but that would be even more complicated.

Reply:
But you can just use `web3.eth.blockNumber` before calling initialize, it doesn't matter to be incorrect at +-1 (or even +- a couple hundreds) blocks.

Reply:
We don't need the block number for the initialize, but for the docker-compose based validator application.